src/Service/TokenService.php

```
@@ -0,0 +1,229 @@
+     */
+    public function getUserFromToken(string $userId): User
+    {
+        $user = $this->userRepository->findByEmail($userId);
+
+        // If not found by email, try by ID (for JWT sub claim)
+        if (null === $user) {
+            $user = $this->userRepository->findOneBy(['id' => $userId]);
+        }
+
+        if (null === $user) {
+            throw new \InvalidArgumentException('User not found');
```

PHPMD (2.15.0)

Missing class import via use statement (line '163', column '23'). (undefined)

(MissingImport)